tools/test-backend

```diff
@@ -152,11 +153,27 @@ def get_failed_tests() -> List[str]:
         print("var/last_test_failure.json doesn't exist; running all tests.")
         return []
 
+def get_failed_thread_tests() -> List[str]:
+    try:
+        with open(FAILED_THREAD_PATH, 'r') as f:
+            return ujson.load(f)
+    except IOError:
+        print("var/failed_thread.json doesn't exist; running all tests.")
+        return []
+
 def write_failed_tests(failed_tests: List[str]) -> None:
     if failed_tests:
         with open(FAILED_TEST_PATH, 'w') as f:
             ujson.dump(failed_tests, f)
 
+def write_failed_thread_tests(failed_thread: List[str]) -> None:
+    if failed_thread:
```

Comment:
I think it'd be better if this conditional lived in the caller, given the function name.